src/coffee/sync/product-sync.coffee

```diff
@@ -37,12 +37,20 @@ class ProductSync extends BaseSync
     super new_obj, old_obj
 
   _doMapActions: (diff, new_obj, old_obj) ->
+    # New variant can have attributes with new values and this can cause conflict if the product still
+    # has old value of the sameForAll attribute. That's why it's necessary to update the attributes
+    # before adding new variant
+    variantActions = @_mapActionOrNot 'variants', => @_utils.actionsMapVariants(diff, old_obj, new_obj)
```

Comment:
https://github.com/sphereio/sphere-node-sdk/pull/198/commits/48688f361068da459f78b9f351763f9b59546614

You need to go up and down anyway ↕️ , but maybe to make it more in line with the code style is a good idea. 👍